app/pathogen/arbovirus/wip/visualizations-section.tsx

```diff
@@ -29,19 +29,24 @@ export const VisualizationsSection = () => {
     ].includes(visualizationInfo.id));
 
     const renderVisualizationList = useCallback((visualizationList: Array<VisualizationInformation & {className: string}>) => {
-      return visualizationList.map((visualization, index) => (
-        <div key={visualization.id} className={cn(visualization.className, index != 0 ? 'mt-14' : undefined)}>
-          <div className="flex py-4">
-            <h3 className="w-full text-center text-lg">{visualization.displayName}</h3>
-            <button
-              onClick={() => router.push(`visualizations?visualization=${visualization.urlParameter}&referrerRoute=/pathogen/arbovirus/wip%3Fsection=${RedesignedArbovirusPageSectionUrlParam.visualizations}`)}
-              aria-label="See visualization in fullscreen"
-            >
-              <ZoomIn />
-            </button>
-          </div>
-          {visualization.renderVisualization()}
-        </div>
+      return visualizationList.map((visualizationInformation) => (
+        <RechartsVisualization
+          key={visualizationInformation.id}
+          visualizationInformation={visualizationInformation}
+          className={cn(visualizationInformation.className, 'pb-14')}
+          buttonConfig={{
+            downloadButton: {
+              enabled: true,
+            },
+            zoomInButton: {
+              enabled: true,
+              referrerRoute: `/pathogen/arbovirus/wip#${RedesignedArbovirusPageSectionId.VISUALIZATIONS}`
```

Comment:
This is why I can't just read the route from NextJS. I want going back to send you to this particular section of WIP. If this isn't working for you it's because the CSS scrolling stuff needs to get merged for this to work but I still think it's a good idea to ship this and then when CSS scrolling stuff is shipped this will start working if that makes sense?